assemblies/client/pom.xml

```diff
@@ -160,6 +160,50 @@
               <excludes>/META-INF/**</excludes>
             </configuration>
           </execution>
+
+          <!-- Add big data driver kar files to karaf's deploy folder -->
```

Comment:
Karaf assemblies should not bring kars in the deploy folder. Kars are meant to be used as a deploy/plugin system when the karaf assembly is already in place. A better alternative is to include the features in the karaf assembly itself when building it. This will make sure that bundles that are present across the multiple kars / features are deployed only once, thus reducing disk footprint. 
Is there any particular why you want to avoid deploying the features and are choosing to deploy Kars instead?

Reply:
@graimundo There are multiple reasons why we're using the deploy folder. Pentaho Hadoop shims don't always ship with the release. We need to be able to drop them into Pentaho products post installation. Also, AEL doesn't use the shims and due to not having a Karaf assembly for AEL, we decided that shims being in the deploy folder would be easier to NOT include.

Reply:
Using the deploy folder isn't a proper production ready solution.

Files dropped in the deploy folder follow a different pipeline of installation (start levels don't apply), sometimes in parallel (prone to errors like https://jira.pentaho.com/browse/BACKLOG-14852 - https://issues.apache.org/jira/browse/KARAF-4723 should fix it), it is less performative (the directory is monitored by Felix fileinstall which "delegates" the file handling to a deployer - e.g. the kar deployer uncompress the kar file, lookups for feature files and install all features in the feature files), it is hard to manage the lifecycle (e.g. you can remove a kar file while karaf is running at it will uninstall, **but** if removed while stopped it will remain there, as it is in cache), you can't control the order between the dropped items, different kars will probably contain duplicate resources in their repositories (adding bloat), ...

Reply:
@nantunes @graimundo The concerns are noted. We've consulted with PO and due to the unique requirements of deploying shims post release and considerations for AEL we're still going to move forward with this strategy for 9.0.